src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,649 @@
+
+const comboBoxOptionStyles = {
+    root: {
+        width: '100%'
+    },
+    flexContainer: { span: { width: '100%' } }
+};
+const freeformOptionsHeaderText = '---';
+const freeformOptionsHeader = {
+    key: 'additional-options',
+    text: freeformOptionsHeaderText,
+    itemType: SelectableOptionMenuItemType.Header
+};
+const getClassNames = classNamesFunction<
+    IResourcePickerStyleProps,
+    IResourcePickerStyles
+>();
+
+const ResourcePicker: React.FC<IResourcePickerProps> = ({
+    adapter,
+    resourceType,
+    requiredAccessRoles,
+    shouldFetchResourcesOnMount = true,
+    displayField = AzureResourceDisplayFields.id,
+    styles,
+    label,
+    loadingLabel,
+    searchParams,
+    onResourcesLoaded,
+    onResourceChange,
+    additionalOptions: additionalOptionsProp,
+    selectedOption: selectedOptionProp,
+    allowFreeform = false,
+    disabled = false
+}) => {
+    const { t } = useTranslation();
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+
+    const [options, setOptions] = useState([]);
+    const [selectedOption, setSelectedOption] = useState<IComboBoxOption>(
+        selectedOptionProp
+            ? {
+                  key: selectedOptionProp,
+                  text: selectedOptionProp,
+                  styles: comboBoxOptionStyles
+              }
+            : null
+    );
+    const [additionalOptions, setAdditionalOptions] = useState<
+        Array<IComboBoxOption>
+    >([]);
+    const [selectedKey, setSelectedKey] = useState<string>(null); // resource id or the option text if manually entered option
+    const [error, setError] = useState(null);
+    const resourcesRef = useRef<Array<IAzureResource>>([]);
+
+    const resourcesState = useAdapter({
+        adapterMethod: () =>
+            adapter.getResourcesByPermissions(
+                resourceType,
+                requiredAccessRoles,
+                searchParams
+            ),
+        refetchDependencies: [adapter, searchParams],
+        isAdapterCalledOnMount: !disabled && shouldFetchResourcesOnMount
+    });
+
+    const placeholder = useMemo(() => {
+        if (displayField === AzureResourceDisplayFields.url) {
+            switch (resourceType) {
+                case AzureResourceTypes.DigitalTwinInstance:
+                    return t('resourcesPicker.enterEnvironmentUrl');
+                case AzureResourceTypes.StorageAccount:
+                    return t('resourcesPicker.enterStorageAccountUrl');
+                case AzureResourceTypes.StorageBlobContainer:
+                    return t('resourcesPicker.enterContainerUrl');
+                default:
+                    return 'resourcesPicker.select';
+            }
+        } else {
+            if (options.length) {
+                return t('resourcesPicker.select');
+            } else {
+                return t('resourcesPicker.noOption');
+            }
+        }
+    }, [resourceType, t, options.length]);
```

Comment:
check you dependency list please. Enable the `"react-hooks/exhaustive-deps": "warn",` rule in your `.eslintrc` and fix all of them in this file.